explorer/interpreter/value.h

```diff
@@ -341,31 +346,36 @@ class NominalClassValue : public Value {
   static constexpr llvm::StringLiteral BaseField{"base"};
 
   NominalClassValue(Nonnull<const Value*> type, Nonnull<const Value*> inits,
-                    std::optional<Nonnull<const NominalClassValue*>> base)
-      : Value(Kind::NominalClassValue),
-        type_(type),
-        inits_(inits),
-        base_(base) {}
+                    std::optional<Nonnull<const NominalClassValue*>> base,
+                    Nonnull<const NominalClassValue** const> class_value_ptr);
 
   static auto classof(const Value* value) -> bool {
     return value->kind() == Kind::NominalClassValue;
   }
 
   template <typename F>
   auto Decompose(F f) const {
-    return f(type_, inits_, base_);
+    return f(type_, inits_, base_, class_value_ptr_);
   }
 
   auto type() const -> const Value& { return *type_; }
   auto inits() const -> const Value& { return *inits_; }
   auto base() const -> std::optional<Nonnull<const NominalClassValue*>> {
     return base_;
   }
+  auto has_vtable() const -> bool { return !vtable().empty(); }
+  auto vtable() const -> const VTable&;
```

Comment:
Can these be removed now?

- I think has_vtable isn't called.
- I think vtable() is only going to be used when the class type is being queried already. As a consequence, getting the vtable from the class type directly doesn't seem like it adds costs.
    - For example, I think value.cpp line 175 uses this, but value.cpp line 179 queries the type so I think it'd be clearer to show both lines accessing the same type.

Reply:
Correct, dropped those.